Wikipedia/Code/WMFFeedContentSource.m

```diff
@@ -227,10 +243,23 @@ - (void)saveGroupForNews:(NSArray<WMFFeedNewsStory *> *)news date:(NSDate *)date
     }
 
     [news enumerateObjectsUsingBlock:^(WMFFeedNewsStory *_Nonnull story, NSUInteger idx, BOOL *_Nonnull stop) {
+        __block unsigned long long mostViews = 0;
+        __block WMFFeedArticlePreview *mostViewedPreview = nil;
         [story.articlePreviews enumerateObjectsUsingBlock:^(WMFFeedArticlePreview *_Nonnull obj, NSUInteger idx, BOOL *_Nonnull stop) {
             NSURL *url = [obj articleURL];
-            [self.previewStore addPreviewWithURL:url updatedWithFeedPreview:obj pageViews:nil];
+            NSDictionary *pageViewsForURL = pageViews[url];
+            NSArray *dates = [pageViewsForURL.allKeys sortedArrayUsingSelector:@selector(compare:)];
```

Comment:
If you set the page views on the preview, there is a method you can use:
`- (NSArray<NSNumber *> *)pageViewsSortedByDate;`